keyboards/gmmk/gmmk2/p65/ansi/keymaps/zen/keymap.c

```diff
@@ -0,0 +1,424 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// For debugging
+#include "print.h"
+
+// Config for this keyboard model
+#include QMK_KEYBOARD_H
+
+// Needed for caps lock animation https://github.com/filterpaper/qmk_userspace/blob/main/rgb/rgb-matrix.c#L72-L75
+#include "lib/lib8tion/lib8tion.h"
+
+// For random
+#include <stdio.h> // for sprintf()
+#include <stdlib.h> // for rand() and srand()
+#include <time.h>   // for time()
+#include "hal.h"  // for MCU access
+#include "analog.h" // to read hardware for entropy
+
+// For leader
+#include <process_leader.h> // for leader
+
+// Custom keys for rgb modes
+enum custom_keycodes {
+    RGB_RCT = SAFE_RANGE, // Set rgb to reactive mode
+    RGB_HTM, // Set RGB to heatmap mode
+    RGB_RBW, // Set RGB to rainbow mod
+    CST_RCH,  // Print a random base64 character
+    CST_RND,  // Generate random string
+};
+
+enum custom_layers {
+    _BL,
+    _FL,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[_BL] = LAYOUT(
+  QK_GESC,  KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,     KC_MINS,  KC_EQL,    KC_BSPC,  KC_DEL,
+  KC_TAB,   KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,     KC_LBRC,  KC_RBRC,   KC_BSLS,  KC_PGUP,
+  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,  KC_QUOT,             KC_ENT,   KC_PGDN,
+  KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,            KC_RSFT,   KC_UP,    KC_END,
+  KC_LCTL,  KC_LGUI,  KC_LALT,                                KC_SPC,	  KC_RALT,  MO(_FL),  KC_LEFT,   KC_DOWN,  KC_RGHT),
+
+[_FL] = LAYOUT(
+  KC_GRV,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,    _______,  KC_INS,
+  CST_RND,  RGB_SAI,  RGB_VAI,  RGB_HUI,  RGB_SPI,  _______,  _______,  _______,  _______,  _______,  KC_PSCR,  KC_SCRL,  KC_PAUS,   KC_MPLY,  KC_RCTL,
+  CST_RCH,  RGB_RMOD, RGB_VAD,  RGB_MOD,  RGB_SPD,  _______,  _______,  _______,  _______,  _______,  _______,  _______,             KC_MUTE,  KC_RCTL,
+  _______,  RGB_HTM,  RGB_RCT,  RGB_RBW,  _______,  QK_BOOT,  _______,  _______,  _______,  QK_LEAD,  KC_MPLY,            _______,   KC_VOLU,  KC_HOME,
+  _______,  _______,  _______,                                 QK_RBT,	  _______,  _______,  KC_MPRV,   KC_VOLD,  KC_MNXT)};
+
+
+// Load leader externals
+LEADER_EXTERNS();
```

Comment:
Leader key was [refactored](https://docs.qmk.fm/#/ChangeLog/20230226?id=leader-key-rework) in recent breaking changes, and should be reflected here.